app/src/androidTest/java/com/github/se/project/ui/EndToEnd.kt

```diff
@@ -0,0 +1,159 @@
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA3").performClick()
+
+    // Click the confirm button
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Go to the profile viewing screen
+    composeTestRule.onNodeWithTag("Profile Icon", true).performClick()
+
+    // Check if the correct profile info is displayed
+    composeTestRule.onNodeWithText("Alice Dupont").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Status: BA3 Student").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Section: SC").assertIsDisplayed()
+
+    // Go to the edit profile screen
+    composeTestRule.onNodeWithTag("editProfileButton").performClick()
+
+    // Change the profile info
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA5").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdown").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdownItem-MA").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Check if the profile info updated correctly
+    composeTestRule.onNodeWithText("Status: BA5 Student").assertIsDisplayed()
+    composeTestRule.onNodeWithText("Section: MA").assertIsDisplayed()
+
+    // Go back to the home screen
+    composeTestRule.onNodeWithTag("closeButton").performClick()
+
+    composeTestRule.onNodeWithText("Find a Tutor").performClick()
+
+    composeTestRule.onNodeWithTag("titleField").performTextInput("End-to-end testing")
+    composeTestRule
+        .onNodeWithTag("DescriptionField")
+        .performTextInput("Teach me how to write tests :(")
+    composeTestRule.onNodeWithTag("DateButton").performClick()
+    onView(withText("OK")).perform(click())
+    composeTestRule.onNodeWithTag("TimeButton").performClick()
+    onView(withText("OK")).perform(click())
+    composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("dropdownAICC").performClick()
+    composeTestRule.onNodeWithTag("mapButton").performClick()
+    composeTestRule.onNodeWithTag("map").performClick()
+    Thread.sleep(2000)
+    val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.click(device.displayWidth / 2, device.displayHeight / 2)
+    composeTestRule.onNodeWithTag("confirmLocation").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    Log.e("abagaga", "profile: ${currentLesson.toString()}")
+    /*assert(currentLesson != null)
+    assert(currentLesson!!.title == "End-to-end testing")
+    assert(currentLesson!!.description == "Teach me how to write tests :(")
+    assert(currentLesson!!.subject == Subject.AICC)
+    assert(currentLesson!!.languages.contains(Language.ENGLISH))*/
```

Comment:
The actual test doesn't choose a tutor (or choose to not choose one) so the lesson is not stored in firebase and not displayed in the overview. I think this is okay for now because this part where we choose tutors will be modified a bit today or tomorrow but it may be usefull then to update as soon as possible the tests so that we choose (or not) a tutor, check if the lesson is correctly stored and check if it appear on the overview.